shopinvader/services/service.py

```diff
@@ -19,6 +19,14 @@ class BaseShopinvaderService(AbstractComponent):
     _collection = "shopinvader.backend"
     _expose_model = None
 
+    @property
+    def env(self):
+        env = self.work.env
+        context = dict(env.context)
+        context["shopinvader_request"] = True
+        env.context = context
```

Comment:
I think you should convert it back to `frozendict`.
Should be something like:
```py
env.context = frozendict(dict(env.context, shopinvader_request=True))
```
Also, I wonder if we should not set this directly in the env of the collection instead, where this env comes from.

Reply:
@simahawk I tried to do it for hours but did not succeed without recreating a new environment.